Framework/API/inc/MantidAPI/IMDHistoWorkspace.h

```diff
@@ -44,6 +44,10 @@ class DLLExport IMDHistoWorkspace : public IMDWorkspace,
   IMDHistoWorkspace_uptr clone() const {
     return IMDHistoWorkspace_uptr(doClone());
```

Comment:
I mean covariance in you actual return type. Subtypes of `IMDHistoWorkspace` must implement a clone that returns a `IMDHistoWorkspace_uptr`. Return a uptr is the right thing to do. I suggest leaving things as they are to avoid unnecessary work.